backend/src/test/java/ee/tlu/forum/controller/ThreadControllerTest.java

```diff
@@ -0,0 +1,266 @@
+    }
+
+    @Test
+    @DisplayName("Returns edited thread - PATCH /api/thread/edit")
+    void editThread() throws Exception {
+        // given
+        User user = new User(1L,
+                "user1",
+                "test1@test.com",
+                "aaa",
+                new ArrayList<>(),
+                "",
+                "",
+                new ArrayList<>(),
+                new ArrayList<>());
+        Thread t1 = new Thread(1L, "content", "title", user, new ArrayList<>());
+
+        Map<String, String> form = new HashMap<>();
+        form.put("title", t1.getTitle());
+        String formJson = new ObjectMapper().writeValueAsString(form);
+
+        when (threadService.editThread(any()))
+                .thenReturn(t1);
+
+        RequestBuilder requestBuilder = MockMvcRequestBuilders.patch("/api/thread/edit")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(formJson);
+
+        // when then
+        mockMvc.perform(requestBuilder)
+                .andExpect(MockMvcResultMatchers.status().is(200))
+                .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.id", Matchers.is(1)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.title", Matchers.is("title")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.text", Matchers.is("content")));
+    }
+
+
+    @Test
+    @DisplayName("Returns all posts of a thread edited thread - GET /api/thread/{id}/posts")
+    void getAllThreadPosts() throws Exception {
+        // given
+        User user = new User(1L,
+                "user1",
+                "test1@test.com",
+                "aaa",
+                new ArrayList<>(),
+                "",
+                "",
+                new ArrayList<>(),
+                new ArrayList<>());
+        Thread t1 = new Thread(1L, "content", "title", user, new ArrayList<>());
+        Post post1 = new Post(1L, "post text", user, t1);
+        Collection<Post> posts = t1.getPosts();
+        posts.add(post1);
+        t1.setPosts(posts);
+
+        when (threadService.getAllPostsByThreadId(any()))
+                .thenReturn(t1.getPosts());
+
+        // when then
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/thread/1/posts"))
+                .andDo(MockMvcResultHandlers.print())
+                .andExpect(MockMvcResultMatchers.status().is(200))
+                .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.jsonPath("$[0].id", Matchers.is(1)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$[0].text", Matchers.is("post text")));
+    }
+
+    @Test
+    @DisplayName("Returns a list of threads belonging to a user by user id - GET /api/threads/userid/{id}")
```

Comment:
Same thing here. Should we perhaps test with a minimum of 2 threads?

Reply:
Sure - added to todo